cmd/app/serve.go

```diff
@@ -110,31 +100,41 @@ var serveCmd = &cobra.Command{
 			log.Logger.Fatal(err)
 		}
 
-		server.EnabledListeners = []string{"http"}
+		decorateHandler := func(h http.Handler) http.Handler {
+			return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+				ctx := r.Context()
 
-		server.ConfigureAPI()
+				// For each request, infuse context with our snapshot of the FulcioConfig.
+				// TODO(mattmoor): Consider periodically (every minute?) refreshing the ConfigMap
+				// from disk, so that we don't need to cycle pods to pick up config updates.
+				// Alternately we could take advantage of Knative's configmap watcher.
+				ctx = config.With(ctx, cfg)
+				ctx = api.WithCA(ctx, baseca)
+				ctx = api.WithCTLogURL(ctx, viper.GetString("ct-log-url"))
 
-		h := server.GetHandler()
-		server.SetHandler(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-			ctx := r.Context()
-
-			// For each request, infuse context with our snapshot of the FulcioConfig.
-			// TODO(mattmoor): Consider periodically (every minute?) refreshing the ConfigMap
-			// from disk, so that we don't need to cycle pods to pick up config updates.
-			// Alternately we could take advantage of Knative's configmap watcher.
-			ctx = config.With(ctx, cfg)
-			ctx = api.WithCA(ctx, baseca)
-			ctx = api.WithCTLogURL(ctx, viper.GetString("ct-log-url"))
-
-			h.ServeHTTP(rw, r.WithContext(ctx))
-		}))
+				h.ServeHTTP(rw, r.WithContext(ctx))
+			})
+		}
 
-		http.Handle("/metrics", promhttp.Handler())
+		prom := http.Server{
+			Addr:    ":2112",
+			Handler: promhttp.Handler(),
+		}
 		go func() {
-			_ = http.ListenAndServe(":2112", nil)
+			_ = prom.ListenAndServe()
 		}()
 
-		if err := server.Serve(); err != nil {
+		port := os.Getenv("PORT")
+		if port == "" {
+			port = "8080"
+		}
```

Comment:
My personal preference would be to leave the `--host` and `--port` command line args implemented.

Reply:
I'll add this back now!

Reply:
Ok these are back, and I've reverted the changes to `deployment.yaml`